internal/controller/scenario/scenario_adapter.go

```diff
@@ -103,48 +104,15 @@ func (a *Adapter) EnsureCreatedScenarioIsValid() (controller.OperationResult, er
 		a.logger.LogAuditEvent("IntegrationTestScenario marked as Valid", a.scenario, h.LogActionUpdate)
 	}
 
-	err := h.AddFinalizerToScenario(a.context, a.client, a.logger, a.scenario, h.IntegrationTestScenarioFinalizer)
-	if err != nil {
-		a.logger.Error(err, "Failed to add finalizer to IntegrationTestScenario")
-		return controller.RequeueWithError(err)
-	}
-
-	return controller.ContinueProcessing()
-}
-
-// EnsureDeletedScenarioResourcesAreCleanedUp is an operation that ensures that all resources related to the
-// deleted IntegrationTestScenario are cleaned up.
-func (a *Adapter) EnsureDeletedScenarioResourcesAreCleanedUp() (controller.OperationResult, error) {
-	if a.scenario.DeletionTimestamp == nil {
-		return controller.ContinueProcessing()
-	}
-
-	environmentsForScenario, err := a.loader.GetAllEnvironmentsForScenario(a.context, a.client, a.scenario)
-	if err != nil {
-		a.logger.Error(err, "Failed to find all Environments for IntegrationTestScenario")
-		return controller.RequeueWithError(err)
-	}
-	for _, testEnvironment := range *environmentsForScenario {
-		testEnvironment := testEnvironment
-		if h.IsEnvironmentEphemeral(&testEnvironment) {
-			dtc, err := a.loader.GetDeploymentTargetClaimForEnvironment(a.context, a.client, &testEnvironment)
-			if err != nil || dtc == nil {
-				a.logger.Error(err, "Failed to find deploymentTargetClaim defined in environment", "environment.Name", &testEnvironment.Name)
-				return controller.RequeueWithError(err)
-			}
-
-			err = h.CleanUpEphemeralEnvironments(a.context, a.client, &a.logger, &testEnvironment, dtc)
-			if err != nil {
-				a.logger.Error(err, "Failed to delete the Ephemeral Environment")
-				return controller.RequeueWithError(err)
-			}
+	// See STONEINTG-815
+	// remove finalizer artifacts
+	if controllerutil.ContainsFinalizer(a.scenario, h.IntegrationTestScenarioFinalizer) {
```

Comment:
I think we'll need to have this code execute before [the check for the deletionTimestamp](https://github.com/konflux-ci/integration-service/blob/4cd09ec6ad79ff1beaace13b2a2930df1e8ec327/internal/controller/scenario/scenario_adapter.go#L59-L61) because we want to remove the finalizer also if the scenario is being deleted.